src/Yaapii.Atoms/IO/InputAsBytes.cs

```diff
@@ -70,6 +70,8 @@ public byte[] AsBytes()
                 { }
                 output = baos.ToArray();
             }
+
+            Dispose();
```

Comment:
Autodispose is a good approach here, but what about we move it to a seperate class, a IInput decorator called SelfClosing?

Maybe also a SelfClosingStream class is needed.

Then we can remove the first using at least. The SelfClosing stream can ensure that the stream will be closed when it has no more data to read.

The SelfClosing Decorator can also accept a Stream in ctor.